src/network/http/websocket.cpp

```diff
@@ -173,6 +154,14 @@ namespace fc { namespace http {
               return _ws_connection->get_request_header(key);
             }
 
+            virtual std::string get_host()override
+            {
+               auto header = get_request_header("BS-Forwarded");
```

Comment:
Header name should be configurable like @abitmore said, and absence of configured value means ignore all headers and use remote host of connection.

Reply:
Implemented a configurable header. Will need to adjust in bitshares-core to make use of it.